services/api-server/tests/unit/test_api__study_workflows.py

```diff
@@ -0,0 +1,318 @@
+        assert self._tmp_path
+        path_to_save = self._tmp_path / f"{file_id}{suffix}"
+        resp = await self._client.get(
+            f"{API_VTAG}/files/{file_id}/content",
+            follow_redirects=True,
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        path_to_save.write_bytes(resp.content)
+        return path_to_save
+
+
+class StudiesTestApi(_BaseTestApi):
+    @_handle_http_status_error
+    async def list_study_ports(self, study_id):
+        resp = await self._client.get(
+            f"/v0/studies/{study_id}/ports",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return OnePage[StudyPort](**resp.json())
+
+    @_handle_http_status_error
+    async def create_study_job(self, study_id, job_inputs: dict) -> Job:
+        resp = await self._client.post(
+            f"{API_VTAG}/studies/{study_id}/jobs",
+            json=jsonable_encoder(job_inputs),
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return Job(**resp.json())
+
+    @_handle_http_status_error
+    async def start_study_job(self, study_id, job_id) -> JobStatus:
+        resp = await self._client.post(
+            f"{API_VTAG}/studies/{study_id}/jobs/{job_id}:start",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return JobStatus(**resp.json())
+
+    @_handle_http_status_error
+    async def inspect_study_job(self, study_id, job_id) -> JobStatus:
+        resp = await self._client.post(
+            f"/v0/studies/{study_id}/jobs/{job_id}:inspect",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return JobStatus(**resp.json())
+
+    @_handle_http_status_error
+    async def get_study_job_outputs(self, study_id, job_id) -> JobOutputs:
+        resp = await self._client.post(
+            f"{API_VTAG}/studies/{study_id}/jobs/{job_id}/outputs",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+        return JobOutputs(**resp.json())
+
+    @_handle_http_status_error
+    async def delete_study_job(self, study_id, job_id) -> None:
+        resp = await self._client.delete(
+            f"{API_VTAG}/studies/{study_id}/jobs/{job_id}",
+            **self._request_kwargs,
+        )
+        resp.raise_for_status()
+
+
+@pytest.fixture
+def input_json_path(tmp_path: Path) -> Path:
+    # https://github.com/wvangeit/osparc-pyapi-tests/blob/master/noninter1/input.json
+    p = tmp_path / "input.json"
+    data = {"f1": 3}
+    p.write_text(json.dumps(data))
+    return p
+
+
+@pytest.fixture
+def input_data_path(tmp_path: Path) -> Path:
+    p = tmp_path / "input.data"
+    data = {"x1": 5.0, "y2": 7.5}
+    p.write_text(json.dumps(data))
+    return p
+
+
+@pytest.fixture
+def test_py_path(tmp_path: Path) -> Path:
+    p = tmp_path / "test.py"
+    code = textwrap.dedent(
+        """\
+    import os
+    import json
+
```

Comment:
Suggestion (but can really be ignored): maybe we can keep this python code in a python file and just read it as text. This way we have type hints and more...